[Rel-5_0] - UnitTest - CustomerRealName - bug 4640

s7design opened the pull request:
Hello @mrcbnsls ,

Hereby is extended unit test for CustomerRealName from the bug #795 . There was an issue when $From needs to be taken from Recipients, param RealName doesn't exists in Recipients. Corrected typo to param Realname and now it perform as expected.

In unit test added 2 more scenarios when there is customer ID and one more with recipient, Additionally included notification tag check, since it differs then auto response.

Please let me know what do you think about this proposal and if there are any questions regarding this PR, feel free to ask.

Regards,
Sanjin

mrcbnsls commented:
Well, great, I think that's what UT are for, right? It has already paid off that you've created it :-D

As some changes were made on the UT, could you please rebase it? Thanks!

s7design commented:
Hi @mrcbnsls 

I updated this PR, I believe it works now. You can see that we solved one issue that occurred in Kernel/System/Ticket/Event/NotificationEvent.pm. If Recipient is not stored in DB ( ie. there is not UserID which is used for $AlreadySent{ $Bundle->{Recipient}->{UserID} } = 1; ) there was message in log  'Use of uninitialized value in hash element at /opt/otrs//Kernel/System/Ticket/Event/NotificationEvent.pm line 303'

Please let me know if we can improve this test.

Regards
Zoran

s7design commented:
Hello @mrcbnsls ,

This PR is rebased and updated.

Regards,
Sanjin

mrcbnsls commented:
Merged, thanks, works good for me. See bugreport for merge URLs.

mrcbnsls commented:
Had to revert this. It seems that under certain circumstances the UT is failing. I could not observe this in my own testing environment though... could you please try to find out more about it?

s7design commented:
Hi @mrcbnsls 

Yes of course, we will check it

Regards
Zoran

mrcbnsls commented:
Any news on this, @s7design? :)

s7design commented:
Hi @mrcbnsls 

Sorry for late answer, we will have in that look.

Regards
Zoran

s7design commented:
Hi @mrcbnsls 

I updated this test, there was a problem with compering some strings. As you can see there is used index() function now. Compering with the operator ~= could make some problems. It looked like True() function doesn't have Name param

```
             $Self->True(
                $ArticleNotification{$NotificationKey} =~ /$Test->{ResultNotification}->{$NotificationKey}/,
                 "Notification: $Test->{Name}, $Test->{ResultNotification}->{$NotificationKey}, tag $NotificationKey"
             );
```
You will see I change $CustomerUser, because it was too long and makes a problem because in some notification such string went in new line so I adjusted this variable to pass the test. IMHO the behavior is the same and the test case is successfully tested.
I think it woks now, actually it works on my system.

Regards
Zoran

mrcbnsls commented:
Thanks, seems to work fine now. Let's see what the UT servers say. Merged in f6af7dd and d1d00e4.
